dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt

```diff
@@ -64,7 +64,8 @@ internal class DataUploadRunnable(
         val batteryFullOrCharging = systemInfo.batteryStatus in batteryFullOrChargingStatus
         val batteryLevel = systemInfo.batteryLevel
         val powerSaveMode = systemInfo.powerSaveMode
-        return (batteryFullOrCharging || batteryLevel > LOW_BATTERY_THRESHOLD) && !powerSaveMode
+        val hasEnoughBattery = (batteryFullOrCharging || batteryLevel > LOW_BATTERY_THRESHOLD)
```

Comment:
That's just nitpicking, but why don't  you do : 
```
val hasEnoughBattery = (systemInfo.externalPowerSource || batteryFullOrCharging || batteryLevel > LOW_BATTERY_THRESHOLD) 
return hasEnoughBattery && !powerSaveMode
```

That way we don't have complex boolean expressions, and IMO being plugged in means you do have enough battery.

Reply:
Also it might be worth a refactor to be more consistent : 
- the `systemInfo.batteryStatus` is of type SystemInfo.BatteryStatus, and we have the list of valid values in the current class
- the `systemInfo.externalPowerSource` is of type boolean and the list of valid values is in the `BroadcastReceiverSystemInfoProvider` class

We should have both list used in the same way in only this or the other class.

Reply:
Because for me `externalPowerSource` is a bit out of the battery level context. It is more saying that no matter if this device has a battery or not right now it is using an external power source.
Regarding suggestion number 2 I am not sure if I understand what you mean. If you meant to make the `externalPowerSource` status as a value of the `SystemInfo.BatteryStatus` I am not really sure we want that as for me `externalPowerSource` is a different information. Or you want instead of the `externalPowerSource` as boolean to have `SystemInfo.PowerSource` enum and to have `WIFI`, `BLUETOOTH`, `AC` and `NOT_PLUGGED` ? I also thought about this but I found it a bit over doing it.

Reply:
For me they both relate to the same thing, if you rename `hasEnoughBattery` to `hasEnoughPower`. 

Regarding solution to, basically we could simply move the computation of `batteryFullOrCharging` to the `BroadcastReceiverSystemInfoProvider` and store that as a boolean in the `SystemInfo` data class

Reply:
Ok for the first point, I guess we could do the suggested for the second point but I would rather name that `hasEnoughPower` because in the case the device is plugged in there is no battery involved.